Update hashchange event to new structure (#14515)
mdn/content: files/en-us/web/api/htmlbodyelement/index.md

_No specific methods; inherits methods from its parent, {{domxref("HTMLElement")}}, and from {{domxref("WindowEventHandlers")}}._

## Events

> The following events are available via `WindowEventHandlers`.

Reviewer: Can we remove mentions of `WindowEventHandlers` throughout? It's a mixin, and referring to it doesn't quite explain why it's possible to listened for an event whose target is `Window` on `HTMLBodyElement`.

The weirdness is shown by this example:
```js
document.body.onhashchange = () => console.log('this callback is called')
document.body.addEventListener('hashchange', () => console.log('this callback is NOT called'))
window.location.hash = '#ha'
```

Only "this callback is called" will be logged.

Maybe we can describe the event handler properties as "aliases" of the same on `Window`, and not that `addEventListener` doesn't work?

Author: This is so weird. wow. Thanks for explaining what is going on here, I had no idea. I tried to reword this in https://github.com/mdn/content/pull/14515/commits/eed1d95b9365097d9a93a4b11d9e1da259e234d9. I hope I understood you correctly.

> The {{domxref("HTMLElement")}} events are inherited and the following {{domxref("Window")}} event handler properties are also available on `HTMLBodyElement` as aliases.

Reviewer 2: So I wasn't able to understand it during the first reading, but @Elchi3 explained it to me.

So I'm proposing two changes:
```suggestion
The {{domxref("HTMLElement")}} events are inherited and the following {{domxref("Window")}} event handler properties (of the `onXYZ` form) are also available on `HTMLBodyElement` as aliases. Even set here, they do apply to the `Window` object.
```

Author: Alternative:

```suggestion
The {{domxref("HTMLElement")}} events are inherited. The following {{domxref("Window")}} `onXYZ` event handler properties are also available as aliases targeting the `window` object. It is advised to call them on the `window` object directly rather than on `HTMLBodyElement`.
```

- {{domxref("window.hashchange_event", "HTMLBodyElement.onhashchange")}}
  - : Fired when the fragment identifier of the URL has changed (the part of the URL beginning with and following the `#` symbol).

> 

Reviewer 2: ```suggestion
> **Note:** Using `addEventListener()` on this object will not work for these events, as they are really dispatched at the {{domxref("window")}} object.
```

Author: Alternative: 

```suggestion
> **Note:** Using `addEventListener()` on `HTMLBodyElement` will not work for the `onXYZ` event handlers listed here. It is advised to use the events on the {{domxref("window")}} object instead.
```

Reviewer 2: I like the alternative with a small nit: _advise to listen to the events_ instead of _to use_.

Author: Yes, thanks! Updated.
